src/libs/actions/IOU.ts

```diff
@@ -1558,6 +1558,8 @@ function getDeleteTrackExpenseInformation(
     const transaction = allTransactions[`${ONYXKEYS.COLLECTION.TRANSACTION}${transactionID}`];
     const transactionViolations = allTransactionViolations[`${ONYXKEYS.COLLECTION.TRANSACTION_VIOLATIONS}${transactionID}`];
     const transactionThreadID = reportAction.childReportID;
+    const chatReportActions = allReportActions?.[`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${chatReportID}`] ?? {};
+    const actionableWhisperReportAction = Object.values(chatReportActions).find((action) => action?.reportActionID === actionableWhisperReportActionID);
```

Comment:
Let us use the utility function [getReportAction](https://github.com/Expensify/App/blob/2026f62b5074d892163f837ca7b030b84c8e350f/src/libs/ReportActionsUtils.ts#L855) to fetch the Whisper action as that would be much neater. Also, don’t we have to fetch the Whisper Action only when whisper action id is available? If so, let us instead fetch this within the if block [here](https://github.com/Expensify/App/blob/2026f62b5074d892163f837ca7b030b84c8e350f/src/libs/actions/IOU.ts#L1709C1-L1721C6)